### ISSUE TITLE
"ERROR: Timed out waiting for TensorBoard to start." on Jupyter

### ISSUE THREAD

**Issue by otavio-silva** [tensorflow/tensorboard#2739]
## Environment information (required)
OS: Windows 10 Education 1903
Python version: 3.7.4
TensorFlow version: 2.0
TensorBoard version: 2.0
Jupyter version: 
`jupyter               1.0.0`
`jupyter-client        5.2.4`
`jupyter-console       6.0.0`
`jupyter-core          4.4.0`
`jupyterlab            1.1.4`
`jupyterlab-server     1.0.0`

Please run `diagnose_tensorboard.py` (link below) in the same
environment from which you normally run TensorFlow/TensorBoard, and
paste the output here:

### Diagnostics

<details>
<summary>Diagnostics output</summary>

``````
--- check: autoidentify
INFO: diagnose_tensorboard.py version 4725c70c7ed724e2d1b9ba5618d7c30b957ee8a4

--- check: general
INFO: sys.version_info: sys.version_info(major=3, minor=7, micro=4, releaselevel='final', serial=0)
INFO: os.name: nt
INFO: os.uname(): N/A
INFO: sys.getwindowsversion(): sys.getwindowsversion(major=10, minor=0, build=18362, platform=2, service_pack='')

--- check: package_management
INFO: has conda-meta: False
INFO: $VIRTUAL_ENV: None

--- check: installed_packages
INFO: installed: tensorboard==2.0.0
INFO: installed: tensorflow-gpu==2.0.0
INFO: installed: tensorflow-estimator==2.0.0

--- check: tensorboard_python_version
INFO: tensorboard.version.VERSION: '2.0.0'

--- check: tensorflow_python_version
INFO: tensorflow.__version__: '2.0.0'
INFO: tensorflow.__git_version__: 'v2.0.0-rc2-26-g64c3d382ca'

--- check: tensorboard_binary_path
INFO: which tensorboard: b'C:\\Python37\\Scripts\\tensorboard.exe\r\n'

--- check: readable_fqdn
INFO: socket.getfqdn(): 'GT62VR-7RD'

--- check: stat_tensorboardinfo
INFO: directory: C:\Users\otavi\AppData\Local\Temp\.tensorboard-info
INFO: os.stat(...): os.stat_result(st_mode=16895, st_ino=8444249301749551, st_dev=3098128582, st_nlink=1, st_uid=0, st_gid=0, st_size=0, st_atime=1570466520, st_mtime=1570466520, st_ctime=1570466517)
INFO: mode: 0o40777

--- check: source_trees_without_genfiles
INFO: tensorboard_roots (1): ['C:\\Python37\\lib\\site-packages']; bad_roots (0): []

--- check: full_pip_freeze
INFO: pip freeze --all:
absl-py==0.7.1

alembic==1.0.8

appdirs==1.4.3

astor==0.8.0

astroid==2.3.1

async-generator==1.10

atomicwrites==1.3.0

attrs==19.1.0

backcall==0.1.0

bleach==3.1.0

boto==2.49.0

boto3==1.9.220

botocore==1.12.220

certifi==2019.3.9

chardet==3.0.4

cloudpickle==0.8.0

colorama==0.4.1

colorclass==2.2.0

cycler==0.10.0

Cython==0.29.13

dask==1.1.4

decorator==4.3.2

defusedxml==0.5.0

doc8==0.8.0

docutils==0.15.2

entrypoints==0.3

gast==0.2.2

gensim==3.8.1

google-pasta==0.1.6

graphviz==0.8.4

grpcio==1.21.1

h5py==2.9.0

idna==2.6

imageio==2.5.0

importlib-metadata==0.23

intel-openmp==2019.0

ipykernel==5.1.0

ipython==7.5.0

ipython-genutils==0.2.0

ipywidgets==7.4.2

isort==4.3.15

jedi==0.15.1

Jinja2==2.10

jmespath==0.9.4

joblib==0.13.2

json5==0.8.4

jsonschema==3.0.1

jupyter==1.0.0

jupyter-client==5.2.4

jupyter-console==6.0.0

jupyter-core==4.4.0

jupyterlab==1.1.4

jupyterlab-server==1.0.0

jupyterthemes==0.20.0

Keras-Applications==1.0.8

Keras-Preprocessing==1.1.0

kiwisolver==1.0.1

lazy-object-proxy==1.4.2

lesscpy==0.13.0

lib==3.0.0

Mako==1.0.7

Markdown==3.1.1

MarkupSafe==1.1.1

matplotlib==3.1.1

mccabe==0.6.1

mistune==0.8.4

mkl==2019.0

mock==3.0.5

more-itertools==7.2.0

mpmath==1.1.0

mxnet-cu100mkl==1.5.0

nbconvert==5.4.1

nbformat==4.4.0

networkx==2.3

nose==1.3.7

notebook==5.7.8

numpy==1.16.5

opencv-python==4.1.1.26

opt-einsum==3.1.0

packaging==19.1

pamela==1.0.0

pandas==0.25.1

pandocfilters==1.4.2

parso==0.5.1

pbr==5.2.0

pickleshare==0.7.5

Pillow==6.2.0

pip==19.2.3

pip-autoremove==0.9.1

pip-check==2.5.2

pip-date==1.0.3

pipdeptree==0.13.2

pluggy==0.13.0

ply==3.11

prometheus-client==0.6.0

prompt-toolkit==2.0.9

protobuf==3.8.0

py==1.8.0

pycodestyle==2.5.0

pygame==1.9.4

Pygments==2.3.1

pylint==2.4.2

pyparsing==2.3.1

pyrsistent==0.14.11

pytest==5.2.0

python-dateutil==2.8.0

python-editor==1.0.4

python-oauth2==1.1.0

pytools==2019.1

pytz==2018.9

PyWavelets==1.0.2

pywinpty==0.5.5

pyzmq==18.0.1

qtconsole==4.4.3

requests==2.18.4

restructuredtext-lint==1.3.0

s3transfer==0.2.1

scikit-image==0.15.0

scikit-learn==0.21.3

scipy==1.3.1

seaborn==0.9.0

Send2Trash==1.5.0

setuptools==41.2.0

six==1.12.0

sklearn==0.0

smart-open==1.8.4

SQLAlchemy==1.3.9

stevedore==1.30.1

sympy==1.4

tensorboard==2.0.0

tensorflow-estimator==2.0.0

tensorflow-gpu==2.0.0

termcolor==1.1.0

terminado==0.8.1

terminaltables==3.1.0

testpath==0.4.2

toolz==0.9.0

torch==1.1.0

torchvision==0.3.0

tornado==6.0.3

traitlets==4.3.2

typed-ast==1.4.0

urllib3==1.22

wcwidth==0.1.7

webencodings==0.5.1

Werkzeug==0.15.4

wget==3.2

wheel==0.33.6

widgetsnbextension==3.4.2

wrapt==1.11.1

xgboost==0.90

xlrd==1.2.0

zipp==0.6.0


``````

</details>

### Next steps

No action items identified. Please copy ALL of the above output,
including the lines containing only backticks, into your GitHub issue
or comment. Be sure to redact any sensitive information.

For browser-related issues, please additionally specify:

  - Browser type and version (e.g., Chrome 64.0.3282.140): Firefox Quantum 69.0.2 (64bit)
  - Screenshot, if it’s a visual issue:

## Issue description

Please describe the bug as clearly as possible. How can we reproduce the
problem without additional resources (including external data files and
proprietary Python modules)?

Just using the code cells below (similar to what is described at [tensorflow.org](https://www.tensorflow.org/tensorboard/tensorboard_in_notebooks))

```
import tensorflow, datetime, os
from tensorflow.keras import applications, datasets
from sklearn import model_selection
```
```
(x_train, y_train), (x_test, y_test) = datasets.cifar10.load_data()
```
```
x_train, x_validation, y_train, y_validation = model_selection.train_test_split(x_train, y_train, test_size=0.25, random_state=42)
```
```
mobilenet = applications.MobileNet(input_shape=x_train.shape[1:], pooling="avg", classes=10, include_top=False)
```
```
mobilenet.compile(tensorflow.optimizers.Adam(learning_rate=0.001), loss="sparse_categorical_crossentropy", metrics=["accuracy"])
log_dir = os.path.join("logs", datetime.datetime.now().strftime("%Y%m%d-%H%M%S"))
tensorboard_callback = tensorflow.keras.callbacks.TensorBoard(log_dir=log_dir, histogram_freq=1)
mobilenet.fit(x_train, y_train, batch_size=128, epochs=10, validation_data=(x_validation, y_validation), callbacks=[tensorboard_callback])
```
```
%load_ext tensorboard
```
```
%tensorboard --logdir logs
```
Gives the output:

![image](https://user-images.githubusercontent.com/22914610/66332313-d12f6600-e90a-11e9-89ca-4b723e4f8247.png)

But when I go to `http://localhost:6006/` the TensorBoard instance is running as seen here:

![image](https://user-images.githubusercontent.com/22914610/66332480-3edb9200-e90b-11e9-898d-20ae3cea4273.png)

And the process does exist as a subprocess of the Jupyter kernel instance:

![image](https://user-images.githubusercontent.com/22914610/66332528-59ae0680-e90b-11e9-8e83-4308c728e9d3.png)

But it's not shown in the Jupyter interface.

**Comment by gowthamkpr** [comment]
Similar to the issue #2057. As mentioned in this [comment](https://github.com/tensorflow/tensorboard/issues/2057#issuecomment-477253676) Can you please create a new environment and try if the error still persists. Thanks!

**Comment by wchargin** [comment]
Sometimes this can happen if the TensorBoard process just takes more
than 10 seconds to start. If you re-run the `%tensorboard` cell with the
same arguments, it should find the existing TensorBoard and attach to it
immediately. Does this work for you, or does it still give an error even
when the TensorBoard process was running prior to running the cell?

**Comment by otavio-silva** [comment]
> 
> 
> Similar to the issue #2057. As mentioned in this [comment](https://github.com/tensorflow/tensorboard/issues/2057#issuecomment-477253676) Can you please create a new environment and try if the error still persists. Thanks!

I tried on a brand new virtual environment, but had no success. This was the output:

![image](https://user-images.githubusercontent.com/22914610/66356478-6b120580-e941-11e9-8dc7-7709afb32b41.png)

But such process with that PID is non-existent. Also, I ran `diagnose_tensorboard.py` on the virtual environment, and this was the output:

### Diagnostics

<details>
<summary>Diagnostics output</summary>

``````
--- check: autoidentify
INFO: diagnose_tensorboard.py version 4725c70c7ed724e2d1b9ba5618d7c30b957ee8a4

--- check: general
INFO: sys.version_info: sys.version_info(major=3, minor=7, micro=4, releaselevel='final', serial=0)
INFO: os.name: nt
INFO: os.uname(): N/A
INFO: sys.getwindowsversion(): sys.getwindowsversion(major=10, minor=0, build=18362, platform=2, service_pack='')

--- check: package_management
INFO: has conda-meta: False
INFO: $VIRTUAL_ENV: 'D:\\otavi\\Documents\\testenv'

--- check: installed_packages
INFO: installed: tensorboard==2.0.0
INFO: installed: tensorflow-gpu==2.0.0
INFO: installed: tensorflow-estimator==2.0.0

--- check: tensorboard_python_version
INFO: tensorboard.version.VERSION: '2.0.0'

--- check: tensorflow_python_version
INFO: tensorflow.__version__: '2.0.0'
INFO: tensorflow.__git_version__: 'v2.0.0-rc2-26-g64c3d382ca'

--- check: tensorboard_binary_path
INFO: which tensorboard: b'D:\\otavi\\Documents\\testenv\\Scripts\\tensorboard.exe\r\nC:\\Python37\\Scripts\\tensorboard.exe\r\n'

--- check: readable_fqdn
INFO: socket.getfqdn(): 'GT62VR-7RD'

--- check: stat_tensorboardinfo
INFO: directory: C:\Users\otavi\AppData\Local\Temp\.tensorboard-info
INFO: os.stat(...): os.stat_result(st_mode=16895, st_ino=8444249301749551, st_dev=3098128582, st_nlink=1, st_uid=0, st_gid=0, st_size=4096, st_atime=1570474020, st_mtime=1570474020, st_ctime=1570466517)
INFO: mode: 0o40777

--- check: source_trees_without_genfiles
INFO: tensorboard_roots (1): ['D:\\otavi\\Documents\\testenv\\lib\\site-packages']; bad_roots (0): []

--- check: full_pip_freeze
INFO: pip freeze --all:
absl-py==0.8.0

astor==0.8.0

attrs==19.2.0

backcall==0.1.0

bleach==3.1.0

colorama==0.4.1

decorator==4.4.0

defusedxml==0.6.0

entrypoints==0.3

gast==0.2.2

google-pasta==0.1.7

grpcio==1.24.1

h5py==2.10.0

ipykernel==5.1.2

ipython==7.8.0

ipython-genutils==0.2.0

jedi==0.15.1

Jinja2==2.10.3

joblib==0.14.0

json5==0.8.5

jsonschema==3.0.2

jupyter-client==5.3.3

jupyter-core==4.5.0

jupyterlab==1.1.4

jupyterlab-server==1.0.6

Keras-Applications==1.0.8

Keras-Preprocessing==1.1.0

Markdown==3.1.1

MarkupSafe==1.1.1

mistune==0.8.4

nbconvert==5.6.0

nbformat==4.4.0

notebook==6.0.1

numpy==1.17.2

opt-einsum==3.1.0

pandocfilters==1.4.2

parso==0.5.1

pickleshare==0.7.5

pip==19.2.3

prometheus-client==0.7.1

prompt-toolkit==2.0.10

protobuf==3.10.0

Pygments==2.4.2

pyrsistent==0.15.4

python-dateutil==2.8.0

pywin32==225

pywinpty==0.5.5

pyzmq==18.1.0

scikit-learn==0.21.3

scipy==1.3.1

Send2Trash==1.5.0

setuptools==41.4.0

six==1.12.0

sklearn==0.0

tensorboard==2.0.0

tensorflow-estimator==2.0.0

tensorflow-gpu==2.0.0

termcolor==1.1.0

terminado==0.8.2

testpath==0.4.2

tornado==6.0.3

traitlets==4.3.3

wcwidth==0.1.7

webencodings==0.5.1

Werkzeug==0.16.0

wheel==0.33.6

wrapt==1.11.2


``````

</details>

### Next steps

No action items identified. Please copy ALL of the above output,
including the lines containing only backticks, into your GitHub issue
or comment. Be sure to redact any sensitive information.


Also, I don't know if this is relevant, but I had a `CUDNN_STATUS_ALLOC_FAILED` when trying to train the model, so I added the cell:

```
devices = tensorflow.config.experimental.list_physical_devices("GPU")
tensorflow.config.experimental.set_memory_growth(devices[0], True)
```

To enable dynamic memory allocation.

**Comment by otavio-silva** [comment]
> 
> 
> Sometimes this can happen if the TensorBoard process just takes more
> than 10 seconds to start. If you re-run the `%tensorboard` cell with the
> same arguments, it should find the existing TensorBoard and attach to it
> immediately. Does this work for you, or does it still give an error even
> when the TensorBoard process was running prior to running the cell?

Thanks for the tip (I also saw your answer from that other topic), I deleted my entire python installation and installed again from zero, tried this and got the same error as above, TensorBoard is trying to access a process that does not exist and I don't know why. But still, if I invoke TensorBoard from command line outside Jupyter it works as if nothing is wrong.

**Comment by JulianSMoore** [comment]
Late to the party and I will not be providing detailed diagnostic information here but FYI I just posted a similar issue here https://stackoverflow.com/questions/59563025/how-to-reset-tensorboard-when-it-tries-to-reuse-a-killed-windows-pid

I confirm that re-running the %tensorboard cell does cause TB to appear after an intitial timeout (and of course it was available at the appropriate URL directly).

I also noted that TB is not handling PIDs properly - referencing dead PIDs (and ports?) and even apparently re-using them (of course it is in fact a different process with new PID, but TB is not picking up on that)

I have attached my minimal ipynb for info as .txt - rename to .ipynb
[Tensorboard Test Minimal - Copy.txt](https://github.com/tensorflow/tensorboard/files/4018548/Tensorboard.Test.Minimal.-.Copy.txt)

NB I'm a tyro at Python and TF so excuse the (lack of) style

The environment was recently rebuilt from scratch now that Anaconda directly supports TF2. Win 64 bit Home, with nvidia 1080Ti card.

**Comment by mthomp89** [comment]
Same issues. Is there a fix coming soon so TensorBoard is more reliable?

**Comment by ramnoob** [comment]
I met same issues . But in my case ,
1st run `tensorboard --inspect --logdir PATH` .
next run `tensorboard --logdir PATH` . 
after that open `http://localhost:6006/` . 
Somehow this ran well in CommandPrompt

**Comment by dulangaheshan** [comment]
in my experience, after timed out error message I run `%reload_ext tensorboard` in the notebook and then it become appears.

**Comment by wchargin** [comment]
We think that we may have fixed this in latest nightlies (#4300). If
you’re affected by this issue, could you please **uninstall** all copies
of TensorBoard, then install `tb-nightly`—

```
pip uninstall -y tensorboard tb-nightly
pip install -U tb-nightly
```

—and see if this fixes the problem?

(You should run these commands from the same virtualenv that you use for
Jupyter notebooks.)

**Comment by wchargin** [comment]
Most people in this thread have indicated in some way that they’re on
Windows (explictly, via Windows fonts, or with Windows-specific commands
in logs), and I don’t see any indications of Linux or Mac. So I’m going
to assume that #4300 covers all these cases.

If you are running on latest `tb-nightly` and still running into an
issue like this, please file a new issue and include all the requested
diagnostic information, and feel free to cross-link this issue. Thanks!

**Comment by wchargin** [comment]
Duplicate of #4300

**Comment by vivek057** [comment]
> Sometimes this can happen if the TensorBoard process just takes more
> than 10 seconds to start. If you re-run the `%tensorboard` cell with the
> same arguments, it should find the existing TensorBoard and attach to it
> immediately. Does this work for you, or does it still give an error even
> when the TensorBoard process was running prior to running the cell?

This worked in my case.. Thanks:)

**Comment by Prasoonjha1** [comment]
> We think that we may have fixed this in latest nightlies (#4300). If you’re affected by this issue, could you please **uninstall** all copies of TensorBoard, then install `tb-nightly`—
> 
> ```
> pip uninstall -y tensorboard tb-nightly
> pip install -U tb-nightly
> ```
> 
> —and see if this fixes the problem?
> 
> (You should run these commands from the same virtualenv that you use for Jupyter notebooks.)

Thanks, It worked

**Comment by beijiguang94** [comment]
> We think that we may have fixed this in latest nightlies (#4300). If you’re affected by this issue, could you please **uninstall** all copies of TensorBoard, then install `tb-nightly`—
> 
> ```
> pip uninstall -y tensorboard tb-nightly
> pip install -U tb-nightly
> ```
> 
> —and see if this fixes the problem?
> 
> (You should run these commands from the same virtualenv that you use for Jupyter notebooks.)

Thanks. It did work for the error with Google Colab.
